### PR TITLE
MHV-49745

### PR DESCRIPTION
**Pull request by gerard-sullivan**
## Summary

 Spec test for prescription_details model (Vets-api/app/models/prescription_details.rb )

## Related issue(s)

https://jira.devops.va.gov/browse/MHV-49745

## Testing done

done unit test locally by running `bundle exec rspec spec/models/prescription_details_spec.rb`

## Screenshots

image (1)

## What areas of the site does it impact?
 unit test coverage on vets-api

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by vmenshutin-bylight — DISMISSED** [review]

**Review by agravell047 — APPROVED** [review]

File: spec/models/prescription_details_spec.rb
Comment (rmtolmach) [inline spec/models/prescription_details_spec.rb:1]: @gerard-sullivan If [app/models/prescription_details.rb](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/models/prescription_details.rb) is the code you're testing, can you write a test for the `sorted_dispensed_date` method?

Comment (gerard-sullivan) [reply spec/models/prescription_details_spec.rb:1]: okay. I will update soon

Comment (gerard-sullivan) [reply spec/models/prescription_details_spec.rb:1]: @rmtolmach I updated. please review again. thanks

Comment (rmtolmach) [reply spec/models/prescription_details_spec.rb:1]: I see you added a test to make sure it has the attribute, but what about the actual sorting? How is that tested? I'd like to see something that checks the conditionals and branching of this code: https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/models/prescription_details.rb#L31-L53

Comment (gerard-sullivan) [reply spec/models/prescription_details_spec.rb:1]: as you can see,  that method is for getting date value of 'sorted_dispensed_date' attribute.
the return value of that method is datetime value that will be assigned to 'sorted_dispensed_date' attribute
@rmtolmach

Comment (rmtolmach) [reply spec/models/prescription_details_spec.rb:1]: @gerard-sullivan my bad, it's not sorting. 
I see `dispensedDate` is nil and so your test is exercising the code around that. What about if it's not `nil`? Can you test that scenario?

Comment (gerard-sullivan) [reply spec/models/prescription_details_spec.rb:1]: @rmtolmach  Update the test for that scenario. Please let me know if you have any question. thanks

Comment (rmtolmach) [reply spec/models/prescription_details_spec.rb:1]: @gerard-sullivan Sorry, I guess I wasn't clear. Can you write more than one test so it hits all possible scenarios?

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**